main.cpp

```diff
@@ -5,24 +5,27 @@
 
 int main()
 {
+    /*Car *car = nullptr;
     std::cout << std::endl << "OPEL" << std::endl;
     PetrolCar opel(new PetrolEngine(120, 1800, 6));
-    opel.accelerate(50);
-    opel.brake();
-    opel.accelerate(-900);
-    opel.refuel();
+    car = &opel;
+    car->accelerate(50);
+    car->brake();
+    car->accelerate(-900);
+    car->restore();
 
     std::cout << std::endl << "NISSAN" << std::endl;
     ElectricCar nissan(new ElectricEngine(130, 650));
-    nissan.charge();
-    nissan.accelerate(80);
-    nissan.engine_ = new ElectricEngine(150, 700);  // Changing an engine during driving is not safe
-    nissan.turnLeft();
+    car = &nissan;
+    car->restore();
+    car->accelerate(80);
+    nissan.setEngine(150, 700);  // Changing an engine during driving is not safe
+    car->turnLeft();
 
     std::cout << std::endl << "TOYOTA" << std::endl;
     HybridCar toyota(new PetrolEngine(80, 1400, 5), new ElectricEngine(100, 540));
-    toyota.accelerate(100);
-    toyota.brake();
-    toyota.charge();
-    toyota.refuel();
+    car = &toyota;
+    car->accelerate(100);
+    car->brake();
+    car->restore();*/
```

Comment:
Is this PR beta version of Cars considering you have commented main program?

Reply:
Comented for clear program Reading, and forget to uncomment it :)

Reply:
done